packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -132,7 +133,11 @@ const DateRangeCalendar = React.forwardRef(function DateRangeCalendar<TDate>(
 
   const ownerState = props;
   const classes = useUtilityClasses(ownerState);
-  const isMobile = React.useContext(WrapperVariantContext) === 'mobile';
+  const wrapperContext = React.useContext(WrapperVariantContext);
+  // defaults to `true` in environments where `window.matchMedia` would not be available (i.e. test/jsdom)
+  // needed as a fallback when this component is used standalone and there is no `WrapperContext`
+  const isDesktop = useMediaQuery('@media (pointer: fine)', { defaultMatches: true });
+  const isMobile = wrapperContext ? wrapperContext === 'mobile' : !isDesktop;
```

Comment:
What do you guys @alexfauquette @flaviendelangle  think? Should we do it?
Currently when `DateRangeCalendar` is used standalone this always resolves to `false` and even in mobile would produce a buggy range previewing behavior... 😢

Reply:
We must just make sure that SSR does not break.
But unless I'm missing a scenario, we are good with your approach.
On the server`rangePreviewDay === null` so `previewingRange === [null, null]`.
And if `previewingRange` is empty, then `isWithinRange`, `isStartOfRange` and `isEndOfRange` returns `false`.
So all the ternary are equal to `false` no-matter what `isMobile` equals on the 1st render.

___

Just, if we are doing that, I would change the ternaries to be `isDesktop ? isXXXX() : false`, to avoid having both `isMobile` and `isDesktop` variables defined.

Reply:
Yeah, it should not impact SSR or tests as it's the similar approach we use for many components.
How would you write this condition differently? 🤔 Am I missing something, but I don't seem to come-up with a more simplistic approach. 🙈 

`isDesktop` here will be used on when this component is used standalone and no `context` is available.
On the other hand—all our internal components having `React.useContext(WrapperVariantContext)` are behaving like this and maybe it's a bit pointless introducing this somewhat pointless optimisation, which does not improve performance in any perceivable way? 🤔

Reply:
> How would you write this condition differently? thinking Am I missing something, but I don't seem to come-up with a more simplistic approach.

Maybe

```tsx
const isDesktopMediaQuery = useMediaQuery('@media (pointer: fine)', { defaultMatches: true });
const isDesktop = wrapperContext ? wrapperContext === 'desktop' : isDesktopMediaQuery
```

My goal is just to avoid using `isDesktop` by mistake (since it would lead to buggy behavior)

Reply:
Fair point. Initially ` : false` (the else) case confused me.

But my question still stands—do you think it's worth introducing this fix/optimisation only to this component, when we have quite a few others, which could also potentially be used without `wrapperContext`.
After thinking about this—I' m leaning towards all or none solution instead of just here. 🙈 🤷

Reply:
> I' m leaning towards all or none solution instead of just here. see_no_evil shrug

I agree, and I'd prefer the "all" solution :angel: 
But that can be in a follow up PR if you want

Reply:
Updated with your suggestion, although, after thinking about it a bit more and checking other cases, where only the `variant` is being passed instead of being resolved into a boolean.
I'm leaning towards removing this edge-case handling—even if just for reducing the computations count. If we were to proceed with changing every case to have this edge-case handling—each component would additionally compute the media query on it's own, event though, it would probably be pointless in 99% of the cases. 🙈

Reply:
I've [created an issue](https://github.com/mui/mui-x/issues/7064) to potentially address this concern in a more appropriate manner.